source/base/data_out_base.cc

```diff
@@ -7560,8 +7560,11 @@ DataOutInterface<dim, spacedim>::write_vtu_with_pvtu_record(
   const unsigned int n_files_written =
     (n_groups == 0 || n_groups > n_ranks) ? n_ranks : n_groups;
 
+  AssertThrow(n_files_written >= 1, ExcMessage("Logical error."));
```

Comment:
If this should be impossible to reach we usually write
```cpp
Assert(n_files_written >= 1, ExcInternalError());
```
Is that the case here?

Reply:
yes, I will change this

Reply:
done in https://github.com/dealii/dealii/pull/8924/commits/20f131f20cfe477e162501d34fece92307858667